File: src/easy/001-two-sum/twoSum.js
var twoSum = function (nums, target) {
  const obj = {};

  for (let i = 0; i < nums.length; i++) {
    if (typeof obj[target - nums[i]] === "number") {
      return [obj[target - nums[i]], i];
    } else {
      obj[nums[i]] = i;
    }
  }

  return [];
};

console.log(twoSum([2, 7, 11, 15], 9));  // output = [0,1]
console.log(twoSum([3, 2, 4], 6));       // output = [1,2]
console.log(twoSum([3, 3], 6));          // output = [0,1]

File: src/easy/001-two-sum/two_sum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        obj = {}

        for i, value in enumerate(nums):
            if target - value in obj:
                return [obj[target - value], i]
            else:
                obj[value] = i

        return []


testSolution = Solution()

print(testSolution.twoSum(nums=[2, 7, 11, 15], target=9))   # output = [0,1]
print(testSolution.twoSum(nums=[3, 2, 4], target=6))        # output = [1,2]
print(testSolution.twoSum(nums=[3, 3], target=6))           # output = [0,1]
